src/plugins/intel_cpu/src/nodes/kernels/registers_pool.hpp

```diff
@@ -0,0 +1,184 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "cpu/x64/jit_generator.hpp"
+#include <dnnl_types.h>
+
+namespace ov {
+namespace intel_cpu {
+
+using namespace dnnl::impl::cpu;
+
+/**
+ * The RegistersPool is the base class for the RegPool template:
+ *      template <x64::cpu_isa_t isa>
+ *      class RegPool : public RegistersPool;
+ *
+ * The registers pool must be created by instantiating the RegPool template, like the next:
+ *      RegPool<isa> regPool { // the list of the registers to be excluded from pool
+ *          Reg64(Operand::RAX), Reg64(Operand::RCX), Reg64(Operand::RDX), Reg64(Operand::RBX),
+ *          Reg64(Operand::RSP), Reg64(Operand::RBP), Reg64(Operand::RSI), Reg64(Operand::RDI)
+ *      };
+ */
+class RegistersPool {
+public:
+    /**
+     * The scoped wrapper for the Xbyak registers.
+     * By creating it you are getting the register from the pool RegistersPool.
+     * It could be created by using constructor with RegistersPool as an argument, like the next:
+     *      const RegistersPool::Reg<Xbyak::Reg64> reg {regPool};
+     * The destructor will return the register to the pool.
+     */
+    template<class TReg>
+    class Reg {
+        friend class RegistersPool;
+    public:
+        Reg(RegistersPool& regPool);
+        ~Reg() { if (onDestruct) { onDestruct(); } }
+        Reg& operator=(Reg&& other)  noexcept {
```

Comment:
Please, make `class Reg` not-copiable and not-movable to prevent `Reg<>` outlive `RegistersPool`, see the following case:
```cpp
auto my_special_op = [this]() {
    IsaRegistersPool<isa> pool{};
    // ...
    RegistersPool::Reg<Xbyak::Reg64> temp{pool};
    // ...
    return temp;
};
```

Reply:
With movable Reg class, the registers could be used more efficiently. For example for two functions with the same functionality funcA will use 2 registers, but funcB needs 3 registers.
``` c++
// function needs at least 2 registers (2 registers must be free in the pool)
    RegistersPool::Reg<Xbyak::Xmm> funcA() {
        RegistersPool::Reg<Xbyak::Xmm> subResult {regPool}; // get 1 XMM register from pool
        // 1 register used
        function_which_use_1_register_from_pool(subReslut); // 2 registers used
        // 1 register used
        RegistersPool::Reg<Xbyak::Xmm> result {regPool}; // get 1 XMM register from pool
        // 2 resister used
        function_which_convert_subresult_to_result(result, subresult);
        return result;
    }

    // usage of funcA
    { // scope for result
        auto result = funcA();
        use_result(result);
    }

    // or
    use_result(funcA());


    // If class Reg will be not-movable the function with same functionality as funcA will look like the next
    // function needs at least 3 registers (1 preallocated and 2 registers must be free in the pool)
    void funcB(Xbyak::Xmm result) {
        // 1 register used
        RegistersPool::Reg<Xbyak::Xmm> subResult {regPool}; // get 1 XMM register from pool
        // 2 resisters used
        function_which_use_1_register_from_pool(subReslut); // 3 registers used
        // 2 registers used
        function_which_convert_subresult_to_result(result, subresult);
        return result;
    }

    // usage of funcB
    { // scope for result
        RegistersPool::Reg<Xbyak::Xmm> result {regPool};
        funcA(result);
        use_result(result);
    }
```